torchgeo/datasets/nccm.py

```diff
@@ -0,0 +1,244 @@
+            checksum: if True, check the MD5 after downloading files (may be slow)
+
+        Raises:
+            AssertionError: if ``years`` or ``classes`` are invalid
+            FileNotFoundError: if no files are found in ``paths``
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+        """
+        assert all(
+            year in self.years for year in years
+        ), f"NCCM data product only exists for the following years: {self.years}"
+
+        assert (
+            set(classes) <= self.cmap.keys()
+        ), f"Only the following classes are valid: {list(self.cmap.keys())}."
+
+        self.paths = paths
+        self.years = years
+        self.classes = classes
+        self.download = download
+        self.checksum = checksum
+        self.ordinal_map = torch.zeros(max(self.cmap.keys()) + 1, dtype=self.dtype)
+        self.ordinal_cmap = torch.zeros((len(self.classes), 4), dtype=torch.uint8)
+
+        curr_path = os.getcwd()
+        curr_path += "/data"
+        if not os.path.exists(curr_path):
+            os.mkdir(curr_path)
+        else:
+            contents = os.listdir(curr_path)
+            for item in contents:
+                item_path = os.path.join(curr_path, item)
+                if os.path.isfile(item_path):
+                    os.remove(item_path)
+                elif os.path.isdir(item_path):
+                    shutil.rmtree(item_path)
+
+        self._verify()
+        super().__init__(paths, crs, res, transforms=transforms, cache=cache)
+
+        for v, k in enumerate(self.classes):
+            self.ordinal_map[k] = v
+            self.ordinal_cmap[v] = torch.tensor(self.cmap[k])
+
+    def __getitem__(self, query: BoundingBox) -> dict[str, Any]:
+        """Retrieve mask and metadata indexed by query.
+
+        Args:
+            query: (minx, maxx, miny, maxy, mint, maxt) coordinates to index
+
+        Returns:
+            sample of mask and metadata at that index
+
+        Raises:
+            IndexError: if query is not found in the index
+        """
+        sample = super().__getitem__(query)
+        sample["mask"] = self.ordinal_map[sample["mask"]]
+        return sample
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset.
+
+        Raises:
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+        """
+        # Check if the extracted files already exist
+        if self.files:
+            return
+
+        # Check if the zip file has already been downloaded
+        assert isinstance(self.paths, str)
+        pathname = os.path.join(self.paths, "**", self.zipfile_glob)
+        if glob.glob(pathname, recursive=True):
+            self._extract()
+            return
+
+        # Check if the user requested to download the dataset
+        if not self.download:
+            raise RuntimeError(
```

Comment:
See #1714, we just changed the error for this kind of situation. Can you rebase/merge main and use this style instead?